libs/blockscout-auth/src/user.rs

```diff
@@ -0,0 +1,85 @@
+use crate::common::{build_http_headers, extract_jwt, CommonError};
+use reqwest::{Client, StatusCode};
+use serde::Deserialize;
+use thiserror::Error;
+use tonic::metadata::MetadataMap;
+use url::Url;
+
+const API_KEY_NAME: &str = "api_key";
+
+#[derive(Debug, Deserialize)]
+pub struct UserInfo {
+    pub address_hash: String,
+    pub avatar: String,
+    pub email: Option<String>,
+    pub name: String,
+    pub nickname: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct ErrorBody {
+    message: String,
+}
+
+#[derive(Error, Debug)]
+pub enum Error {
```

Comment:
why do you need second enum Error? this is like the same thing as auth::Error. Also you dont export this Error in lib.rs so it's not possible to import it from the crate.

Reply:
this is small crate, there is no need to create 2 errors in move "common" into CommonError. also Common is bad naming because it can mean anything